test_gen_scripts/utils.py

```diff
@@ -2,10 +2,27 @@
 import string
 from hashlib import sha256
 from ecdsa import SigningKey, NIST256p, util
+import os
 
+def generate_deterministic_key(seed=None):
+    if seed is not None:
+        rng = DeterministicPRNG(seed)
+    else:
+        rng = os.urandom
 
-def get_good_signature(message: bytes, hash=False):
-    sk = SigningKey.generate(curve=NIST256p)
+    sk = SigningKey.generate(curve=NIST256p, entropy=rng)
+    return sk
+
+class DeterministicPRNG:
+    def __init__(self, seed):
+        self.seed = seed
+
+    def __call__(self, n):
+        hash_value = sha256(self.seed).digest()
+        return hash_value[:n]
+
+def get_good_signature(message: bytes, hash=False, seed=None):
```

Comment:
Revert the signature change so that the end user doesn't have to work with the seed:
`def get_good_signature(message: bytes, hash=False):`